content/en/platform/corda/5.2/deploying-operating/deployment/upgrading521/_index.md

```diff
@@ -0,0 +1,517 @@
+
+2. Verify the generated SQL scripts and apply them to the PostgreSQL database. For example:
+
+   ```shell
+   psql -h localhost -p 5432 -f ./sql_updates/config.sql -d cordacluster -U postgres
+   ```
+
+3. Grant the necessary permissions. For example:
+
+   ```shell
+   psql -h localhost -c "GRANT SELECT, INSERT, UPDATE, DELETE ON ALL TABLES IN SCHEMA config TO corda" -p 5432 -d cordacluster -U postgres
+   psql -h localhost -c "GRANT USAGE, SELECT ON ALL SEQUENCES IN SCHEMA config TO corda" -p 5432 -d cordacluster -U postgres
+   ```
+
+   The `cluster` database user is the user set up by the Corda Helm chart when the configuration database was originally deployed.
+
+4. Apply the SQL files generated in step [Migrate Data From Kafka Topics to the Cluster Database](#migrate-data-from-kafka-topics-to-the-cluster-database).
+
+   {{< note >}}
+   You must set the schema to the cluster configuration schema when using `psql`. By default, this is the configuration as used in this example. This is because the migration script is schema-agnostic.
+   {{< /note >}}
+
+   ```shell
+   PGOPTIONS='-c search_path=config' psql -h localhost -p 5432 -f ./sql_updates/migrate_member_data.sql -d cordacluster -U postgres
+   ```
+
+## Optional: Clear Down Key Rotation Stale Data
+
+Due to a fix in version 5.2.1, the transient flow status state has moved from the key rotation state manager schema to the flow state manager schema. As a result, the automatic cleanup of this data in the old key rotation status manager schema does not occur. Corda 5.2.1 cleans it only from the new location, potentially leaving stale data in the key rotation schema. To avoid unnecessary storage of stale data, R3 recommends that you delete all states from the key rotation state manager database.
+
+The only functional consequence of deleting the key rotation state data is the loss of legacy key rotation status information. This does not affect the outcome of those key rotations but only affects any subsequent attempts to report their status. Using the key rotation API to check the status of past rotations will return a result as if they were never completed. Note that Corda only retains the most recent key rotation status for any tenant.
+
+If you need legacy key rotation data, you can omit this step or perform it later. The upgrade process does not require it. If you choose to perform it later, ensure no key rotation operations are running at that time.
+
+The name of the key rotation schema depends on the value `stateManager.keyRotation.partition` provided to the Corda Helm chart during bootstrapping. If not specified, it defaults to `sm_key_rotation`.
+
+To delete the key rotation state, assuming the default schema name, use:
```

Comment:
We should make it clear this is just an example of the use of `psql` like the ones above.